js/README.md

```diff
@@ -0,0 +1,275 @@
+
+## Arrays (vectori)
+
+Este un tip de date care este foarte folosit si e asemanator cu cel din limbajele standard.
+
+```js
+// ARRAYS - Store multiple values in a variable
+const numbers = [1,2,3,4,5];
+const fruits = ['apples', 'oranges', 'pears', 'grapes'];
+console.log(numbers, fruit);
+
+// Get "oranges" value - Arrays start at 0
+console.log(fruits[1]);
+
+// Add value
+fruits[4] = 'blueberries';
+
+// Add value using push()
+fruits.push('strawberries');
+
+// Add to beginning
+fruits.unshift('mangos');
+
+// Remove last value
+fruits.pop();
+
+// Get index
+console.log(fruits.indexOf('oranges'));
+```
+
+## Objects (JSON)
+
+Obiectele reprezinta o modalitate foarte populara de a organiza datele (exista o intreaga paradigma in jurul lor, se invata in anul 2 la materia POO). Acestea sunt prezente si in JS, ba chiar mai mult, au definit un standard international de comunicare intre servicii: [JSON](https://en.wikipedia.org/wiki/JSON).
+
+```js
+const person = {
+  firstName: 'John',
+  age: 30,
+  hobbies: ['music', 'movies', 'sports'],
+  address: {
+    street: '50 Main st',
+    city: 'Boston',
+    state: 'MA'
+  }
+}
+
+// Get single value
+console.log(person.name)
+
+// Get array value
+console.log(person.hobbies[1]);
+
+// Get embedded object
+console.log(person.address.city);
+
+// Add property
+person.email = 'jdoe@gmail.com';
+```
+
+## Conditionals
+- If
+    ```js
+    const x = 30;
+
+    if(x === 10) {
+        console.log('x is 10');
+    } else if(x > 10) {
+        console.log('x is greater than 10');
+    } else {
+        console.log('x is less than 10')
+    }
+    ```
+- Switch
+    ```js
+    color = 'blue';
+
+    switch(color) {
+    case 'red':
+        console.log('color is red');
+    case 'blue':
+        console.log('color is blue');
+    default:  
+        console.log('color is not red or blue')
+    }
+    ```
+- Ternary Operator (shorthand if)
+    ```js
+    const z = color === 'red' ? 10 : 20;
+    ```
+
+**Observatie**. ```===``` fata de ```==``` verifica si tipul variabilei , nu doar continutul.
+- ``` "10" == 10 ``` -> true
+- ``` "10" === 10 ``` -> false
+
+## Loops
+- For
+    ```js
+    for(let i = 0; i <= 10; i++){
+    console.log(`For Loop Number: ${i}`);
+    }
+    ```
+- While
+    ```js
+    let i = 0
+    while(i <= 10) {
+        console.log(`While Loop Number: ${i}`);
+        i++;
+    }
+    ```
+- Parcurgeri
+    ```js
+    for(let i = 0; i < arr.length; i++){
+        console.log(arr[i]);
+    }
+    ```
+    ```
+    for(let element of arr) {
+        console.log(element);
+    }
+    ```
+
+## Functii
+```js
+function greet(name) {
+    return `Hello, ${name}!`;
+}
+
+console.log(greet("Mihai"));
+```
+
+Interesting topic: [Javascript Closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures).
+
+## High Order Array Functions
+De obicei destul de uzuale cand lucram cu array-uri, ele deriva din Programarea Functionala, o alta paradigma (JS e un limbaj multi-paradigm).
```

Comment:
Majoritatea limbajelor sunt multi-paradigm, dar JS la baza lui e functional (si poti sa remarci ca toate celelalte sunt de fapt derivari, in cazul de fata).
Nu as lasa paranteza aia, intrucat ar putea induce in eroare, but your call